CONTRIBUTING.md

```diff
@@ -118,7 +118,62 @@ publicly on the [appropriate chat room][gi] on Gitter asking for a review. Once
 your code has been reviewed, revised if necessary, and then signed-off by a
 developer, it will be merged into the source tree.
 
+### Dealing with upstream changes.
+
+When pulling remote changes to a local branch/machine, we recommend users to rebase instead of
+creating merge commits. 
+
+This is used sometimes when an upstream change cause problems with your pull request. The best practice is
+to do a fast-forward(ff) rebase.
+
+First setup a remote called `upstream`.
+
+```bash
+# Do one of the following. ssh is prefered, but not available on all
+# environments.
+
+# For ssh
+git remote add upstream git@github.com:amethyst/amethyst.git
+# For https
+git remote add upstream https://github.com/amethyst/amethyst.git
+```
+
+If your `origin` remote points to the original repo we recommend you to set it
+to your own fork. (check with `git remote origin get-url`)
+
+```bash
+# Set origin remote to fork, <your-fork> is git@github.com:<your-username>/amethyst.git
+# For https use https://github.com/<your-username>/amethyst.git
+git remote origin set-url <your-fork>
+```
+
+To learn how to rebase a upstream change into your branch read [this excellent wiki][rebase-how-to]
+
+TL;DR
+```bash
+# Fetch latest changes
+git fetch upstream
+# Rebase unto branch
+git checkout <branch-name>
```

Comment:
Should this be "# Rebase this branch" or maybe "# Rebase this branch to upstream"?